### PR TITLE
terraform with modules

### PR DESCRIPTION
**Pull request by nexo8937**
creating terraform modules, making remote state file with s3bucket, updating Readme, changing packer file name

### CODE REVIEW

File: terraform/main/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ value = module.database.rds_endpoint
+}
+
+output "image-sg-id"  {
+ value = module.securitygroup.image-sg
+}
+
+output "autoscaling-id"  {
+ value = module.securitygroup.autoscaling-sg
+}
```
Comment (rubentsirunyan) [inline terraform/main/outputs.tf:19]: Why is this output called `autoscaling-id`?

File: terraform/main/modules.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  vpc    = module.network.vpc
+}
+
+#aws loadbalancer module
+module "loadbalancer" {
+  source = "git@github.com:nexo8937/terraform-module-loadbalancer.git"
+  pub-sub-A = module.network.pub-subnet-A
+  pub-sub-B = module.network.pub-subnet-B
+  lb-sg     = module.securitygroup.lb-sg
+}
```
Comment (rubentsirunyan) [inline terraform/main/modules.tf:27]: What is the reason for having the `loadbalancer` module in `terraform/main` and not `terraform/autoscaling`?

**Review by rubentsirunyan — CHANGES_REQUESTED** [review]
As you committed your modules directly to the main branches without PRs, I'll add my comments here:
1. Please remove all the `exo`s from the resource names. The module resource names should be as generic as possible.
2. In the `autosacling` module:
    2.1. The [name is hardcoded](https://github.com/nexo8937/terraform-module-autoscalling/blob/d3d7668cd60137754c27fb730ed9a73ff71619c4/autoscalling.tf#LL3C26-L3C26). What if someone wants to use your module for something not related to Wordpress?
    2.2. I suggest having the [instance_type](https://github.com/nexo8937/terraform-module-autoscalling/blob/d3d7668cd60137754c27fb730ed9a73ff71619c4/autoscalling.tf#LL6C3-L6C4) as a variable, which can have a default value of `t2.micro` if you want.
    2.3. I suggest having the scale up/down thresholds and periods as variables.
3. In the `database` module, almost all the parameters like db name, username, password and etc. should be provided as variables and not hardcoded.
4. Please remove all the `wordpress` references in the `network` module.
5. What is the reason for having a security group module? That module is not reusable at all. It would be much better to move each security group to its corresponding module, like rds to `database`, `lb-sg` to `loadbalancer` and etc.
6. In the `loadbalancer` module please create variables for all the options that you think the users might want to change.

**Comment by nexo8937** [comment]
dear Ruben, changes was done, please recheck

**Review by rubentsirunyan — APPROVED** [review]

**Pull request merged**